app/frontend/src/utils/transformUtils.js

```diff
@@ -64,3 +65,147 @@ export function attachAttributesToLinks(formSchemaComponents) {
     }
   });
 }
+
+
+/**
+ * This function could be removed from frontend in the next commits
+ * @function isEligibleLateSubmission
+ * Get All possible dates in given period with Term and Interval
+ *
+ * @param {Object[]} date An object of Moment JS date
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @returns {Boolean} Return true if form is available for late submission
+ */
+export function isEligibleLateSubmission(date,term,interval){
+  var gracePeriodDate = moment(date,'YYYY-MM-DD HH:mm:ss').add(term,interval).format('YYYY-MM-DD HH:mm:ss');
+  var isBetweenClosrAndGraceDate = moment().isBetween(date, gracePeriodDate);
+  return isBetweenClosrAndGraceDate;
+}
+
+/**
+ * @function getAvailableDates
+ * Get All possible dates in given period with Term and Interval
+ *
+ * @param {Integer} keepAliveFor An integer for number of days
+ * @param {String} keepAliveForInterval A string of days,Weeks,months
+ * @param {Object[]} substartDate An object of Moment JS date
+ * @param {Integer} term An integer of number of Days/Weeks OR Years
+ * @param {String} interval A string of days,Weeks,months
+ * @param {Integer} allowLateTerm An integer of number of Days/Weeks OR Years
+ * @param {String} allowLateInterval A string of days,Weeks,months
+ * @param {Object[]} repeatUntil An object of Moment JS date
+ * @param {String} scheduleType A string one of Manual, ClosingDate OR Period
+ * @param {Object[]} closeDate An object of Moment JS date
+ * @returns {Object[]} An object array of Available dates in given period
+ */
+export function getAvailableDates(
+  keepAliveFor=0,
+  keepAliveForInterval='days',
+  submstartDate,
```

Comment:
Why is this submstartDate when the documentation says it's supposed to be substartDate?